packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -157,6 +166,16 @@ export interface IFluidContainer<TContainerSchema extends ContainerSchema = Cont
 	 */
 	readonly attachState: AttachState;
 
+	/**
+	 * The read-only information about the container.
+	 *
+	 * @remarks
+	 *
+	 * This is used to determine if the container is read-only or not.
+	 */
+	// TODO: yunho: should we mark it as optional?
+	readonly readOnlyInfo: ReadOnlyInfo;
```

Comment:
ReadOnlyInfo is a somewhat confusing type, especially for customers of IFluidContainer who don't have concepts like storageOnly.  It is also asymmetric to the "readonly" event you're adding in this PR which only parameterizes readonly state as a boolean.

I'd recommend either:
1. If boolean is sufficient for the target scenarios, prefer to just expose this as a boolean readOnly property to match the eventing story and simplify for the customer.
2. If more details about the reason for readonly are really needed for target scenarios (i.e. forced, permissions), consider designing a scoped type that will wholly make sense for an IFluidContainer customer.  And consider whether the "readonly" boolean event parameter should be omitted, changed to match, etc.

Reply:
FYI, this mismatch comes from IContainer level. In IContainer, it has getter's return type as ReadonlyInfo while the eventhandler call back only has boolean. I am not sure if I should fix it in IContainer level as well.

Reply:
Yep - I'm saying that is probably a confusing choice there too :) Though no need to try and fix it for IContainer in the short term.

In general the customer of our `@public` APIs are completely distinct from customers of our `@legacy` APIs like IContainer.  We don't need to mimic `@legacy` API behaviors (including getter return types) on these `@public` APIs, but should instead design based on what would make the most sense to an `@public` customer.

Reply:
After talking with Vlad, we agree that the readOnly return type can be boolean. And we can add more info to ODSPService later if client needs more.